src/redux/wallets.js

```diff
@@ -112,13 +124,46 @@ export const walletsSetSelected = wallet => async dispatch => {
   });
 };
 
-export const isCreatingAccount = val => dispatch => {
+export const setIsWalletLoading = val => dispatch => {
   dispatch({
     payload: val,
-    type: WALLETS_SET_IS_CREATING_ACCOUNT,
+    type: WALLETS_SET_IS_LOADING,
   });
 };
 
+export const setWalletBackedUp = (
+  walletId,
+  method,
+  backupFile = null
+) => async (dispatch, getState) => {
+  const { wallets, selected } = getState().wallets;
+  const newWallets = { ...wallets };
+  newWallets[walletId].backedUp = true;
+  newWallets[walletId].backupType = method;
+  if (backupFile) {
+    newWallets[walletId].backupFile = backupFile;
+  }
+  newWallets[walletId].backupDate = Date.now();
+  dispatch(walletsUpdate(newWallets));
+  if (selected.id === walletId) {
+    dispatch(walletsSetSelected(newWallets[walletId]));
+  }
+
+  // Reset the loading state 1 second later
+  setTimeout(() => {
+    dispatch(setIsWalletLoading(null));
+  }, 1000);
+
+  if (method === WalletBackupTypes.cloud) {
+    try {
+      await backupUserDataIntoCloud({ wallets: newWallets });
+    } catch (e) {
+      logger.error('SAVING WALLET USERDATA FAILED', e);
+    }
+  }
+  await saveUserBackupState(BackupStateTypes.done);
```

Comment:
Perhaps we should think about whether this should be considered "done" if the method was "cloud" backup and it failed above

Reply:
100% agree. See previous comment ^^